torchdynamo/eval_frame.py

```diff
@@ -188,21 +192,31 @@ def _fn(*args, **kwargs):
 
 
 class OptimizeContext(_TorchDynamoContext):
-    def __init__(self, callback, backend_ctx_ctor):
+    def __init__(
+        self, callback, backend_ctx_ctor, pre_on_enter=nothing, post_on_exit=nothing
+    ):
         def on_enter():
+            pre_on_enter()
             global most_recent_backend
             if (
                 most_recent_backend is not None
                 and most_recent_backend is not compiler_fn
             ):
                 raise ResetRequired()
             most_recent_backend = compiler_fn
+            global most_recent_context
+            most_recent_context = backend_ctx_ctor
             install_generation_tagging_init()
 
+        def on_exit():
+            # Could just pass post_on_exit through, but this reads much clearer
```

Comment:
Let's remove this extra wrapper.  It will just add overhead.

Reply:
sg